batch/batch/driver/instance_pool.py

```diff
@@ -135,7 +135,7 @@ def adjust_for_add_instance(self, instance):
 
         self.instances_by_last_updated.add(instance)
         if instance.state in ('pending', 'active'):
-            self.live_free_cores_mcpu += instance.free_cores_mcpu
+            self.live_free_cores_mcpu += max(0, instance.free_cores_mcpu)
```

Comment:
What's this for?

Reply:
free cores can be negative, but the scheduler uses live_free_cores and that shouldn't be negative. In retrospect, maybe you're right and this is fine. But it definitely needs the max when we're subtracting from the live free cores.

I just thought it was better for it to be bounded at 0 for this logic:

```python3
instances_needed = ((ready_cores_mcpu - self.live_free_cores_mcpu + (self.worker_cores * 1000) - 1) // (self.worker_cores * 1000))
```

Reply:
But maybe it can be negative.